Sources/Extensions/SwiftStdlib/DoubleExtensions.swift

```diff
@@ -30,6 +30,27 @@ public extension Double {
     }
     #endif
 
+    /// SwifterSwift: Returns a rounded value with the specified number of
+    /// decimal places and rounding rule. If `numberOfDecimalPlaces` is negative,
+    /// `0` will be used.
+    ///
+    ///     let num: Double = 3.1415927
+    ///     num.rounded(numberOfDecimalPlaces: 3, rule: .up) -> 3.142
+    ///     num.rounded(numberOfDecimalPlaces: 3, rule: .down) -> 3.141
+    ///     num.rounded(numberOfDecimalPlaces: 2, rule: .awayFromZero) -> 3.15
+    ///     num.rounded(numberOfDecimalPlaces: 4, rule: .towardZero) -> 3.1415
+    ///     num.rounded(numberOfDecimalPlaces: -1, rule: .toNearestOrEven) -> 3
+    ///
+    /// - Parameters:
+    ///   - numberOfDecimalPlaces: The expected number of decimal places.
+    ///   - rule: The rounding rule to use.
+    /// - Returns: The rounded value.
+    func rounded(numberOfDecimalPlaces: Int, rule: FloatingPointRoundingRule) -> Double {
+        let number = numberOfDecimalPlaces < 0 ? 0 : numberOfDecimalPlaces
+        let factor = pow(10.0, Double(number))
```

Comment:
You can avoid the line above with
`let factor = pow(10.0, Double(max(0, number)))`